integration-tests/features/steps/stack_analyses_v2.py

```diff
@@ -234,6 +271,157 @@ def verify_vulnerability_count(context, component, vulnerability, expected):
+
+
+def verify_valid_data_companion(context, component, latest_version):
+    """Check valid data in companion packages."""
+    json_data = get_json_data(context)
+    check_attribute_presence(json_data, 'recommendation')
+
+    is_component_present = False
+    for dep in json_data['recommendation']['companion']:
+        if dep['name'] == component:
+            is_component_present = True
+            actual_version = dep['latest_version']
+            assert actual_version == latest_version, \
+                'Component {} have {} but expected {}'.format(component, actual_version,
+                                                              latest_version)
+    assert is_component_present, 'Component not present'
+
+
+def check_essentials(vuln):
+    """Check different feilds that are essential."""
+    assert "id" in vuln
+    assert "cvss" in vuln
+    assert "url" in vuln
+    assert "cwes" in vuln
+    assert "title" in vuln
+    assert "cve_ids" in vuln
+    assert "cvss_v3" in vuln
+    assert "severity" in vuln
+
+
+def verify_license_data(context, component, license_c):
+    """Licenses check."""
+    json_data = get_json_data(context)
+    check_attribute_presence(json_data, 'analyzed_dependencies')
+    is_component_present = False
+    for dep in json_data['analyzed_dependencies']:
+        if dep['name'] == component:
+            is_component_present = True
+            assert "licenses" in dep, 'Licenses not found'
+            licenses = dep["licenses"]
+            for lic in licenses:
+                if lic == license_c:
+                    assert lic == license_c
+    assert is_component_present, 'Component not present'
+
+
+def verify_severity_analyzed_deps(context, component, title, severity, vulenerability):
+    """Check valid vulenerability title and severity."""
+    json_data = get_json_data(context)
+    check_attribute_presence(json_data, 'analyzed_dependencies')
+    is_component_present = False
+    for dep in json_data['analyzed_dependencies']:
+        if dep['name'] == component:
+            is_component_present = True
+            if vulenerability == 'public_vulnerabilities':
+                for vuln in dep['public_vulnerabilities']:
+                    if vuln['title'] == title:
+                        assert vuln['severity'] == severity, "severity not matched"
+            elif vulenerability == 'private_vulnerabilities':
+                for vuln in dep['private_vulnerabilities']:
+                    if vuln['title'] == title:
+                        assert vuln['severity'] == severity, "severity not matched"
+    assert is_component_present, 'Component not present'
+
+
+def github_checks(context):
+    """Check all the analyzed deps for github object."""
+    json_data = get_json_data(context)
+    check_attribute_presence(json_data, 'analyzed_dependencies')
+    for dep in json_data['analyzed_dependencies']:
+        assert "github" in dep, "github data not found for dependency"
```

Comment:
Can we create a separate function to check for all github parameters?
Also, I recommend modifying existing function check_dependency_attributes() to use this new function and add validation for 'github' attribute.

Reply:
have modified the function please have a look